src/main.js

```diff
@@ -1,32 +1,61 @@
+const fs = require('fs')
+const path = require('path')
 const spawn = require('child_process').spawn;
-const app = require('electron').app
-const BrowserWindow = require('electron').BrowserWindow
+const { app, BrowserWindow, screen } = require('electron')
 const fetch = require('node-fetch')
 
-// Keep a global reference of the window object, if you don't, the window will
-// be closed automatically when the JavaScript object is garbage collected.
-let mainWindow;
-
-const isDevMode = process.execPath.match(/[\\/]electron/);
-if (isDevMode) {
+const isDevMode = function() {
+  return process.argv[2] == '--dev'
+};
+if (isDevMode()) {
   // load the '.env' file from the project root
   const dotenv = require('dotenv');
   dotenv.config();  
 }
 
-let PYTHON = (process.env.PYTHON || 'python').trim();
+// Binding to the invest server binary:
+let serverExe;
+
+// A) look for a local registry of available invest installations
+const investRegistryPath = path.join(
+  app.getPath('userData'), 'invest_registry.json')
+if (fs.existsSync(investRegistryPath)) {
+  const investRegistry = JSON.parse(fs.readFileSync(investRegistryPath))
+  const activeVersion = investRegistry['active']
+  serverExe = investRegistry['registry'][activeVersion]['server']
+
+// B) check for dev mode and an environment variable from dotenv
+} else if (isDevMode()) {
+  serverExe = process.env.SERVER
+
+// C) point to binaries included in this app's installation.
+} else {
+  const binary = (process.platform === 'win32') ? 'server.exe' : 'server'
+  // serverExe = path.join(__dirname, 'invest', binary)
+  console.log(process.resourcesPath)
+  serverExe = path.join(
+    process.resourcesPath, 'app.asar.unpacked', 'build', 'invest', binary)
+  console.log(serverExe)
+}
```

Comment:
This bit of InVEST binary loading looks pretty similar to what's in `src/index.js` and seems like an atomic operation, too ... would it be worth refactoring into a common place?

Reply:
Very true. Instead of factoring out to a function in `utils.js` and calling it in both places, I changed this so `main.js` finds both the server and invest exe files (since they're always together) and then passes the necessary data to the renderer process (`index.js` and the rest of the app) via IPC. I bet that IPC could turn up useful for other stuff too.